test/core/miner/precommit_batcher_test.cpp

```diff
@@ -74,8 +83,38 @@ namespace fc::mining {
         return ERROR_TEXT("ERROR");
       };
 
+      std::shared_ptr<FeeConfig> fee_config = std::make_shared<FeeConfig>();
+      fee_config->max_precommit_batch_gas_fee.base = static_cast<TokenAmount>(
+          0.05
+          * 10e18);  // TODO: config loading;
+      fee_config->max_precommit_batch_gas_fee.per_sector =
+          static_cast<TokenAmount>(0.25 * 10e18);
+      MinerInfo miner_info{.control = {side_address_, miner_address_}};
       batcher_ = std::make_shared<PreCommitBatcherImpl>(
-          std::chrono::seconds(60), api_, miner_address_, scheduler_);
+          std::chrono::seconds(60), api_, miner_address_, scheduler_, [=](MinerInfo miner_info,
+                                                                             TokenAmount deposit,
+                                                                             TokenAmount good_funds) -> outcome::result<Address> {
+            Address minimal_balanced_address;
+            auto finder = miner_info.control.end();
+            for (auto address = miner_info.control.begin();
+                 address != miner_info.control.end();
+                 ++address) {
+              if (api_->WalletBalance(*address).value() >= good_funds
+                  && (finder == miner_info.control.end()
+                      || api_->WalletBalance(*finder).value()
+                         > api_->WalletBalance(*address).value())) {
+                finder = address;
+              }
+            }
+            if (finder == miner_info.control.end()) {
+              return miner_info.worker;
+            }
+            if(*finder == side_address_){
+              side_address_called = true;
+            }
+            return *finder;
+          },
```

Comment:
Duplicate. Move to new `.hpp` file and reuse.
Either make `struct` with `res operator()(arg1, arg2, arg3)` and members captured in lambda, or `inline` (to prevent duplicate symbols link error) function which returns this lambda.

Or mock/simplify if this test is not about "address selector".

Reply:
fixed